jcapi-systems.go

```diff
@@ -0,0 +1,334 @@
+		system.Id = fields["_id"].(string)
+
+	}
+	if _, exists := fields["lastContact"]; exists {
+		system.LastContact = fields["lastContact"].(string)
+	}
+	if _, exists := fields["remoteIP"]; exists {
+		system.RemoteIP = fields["remoteIP"].(string)
+	}
+	if _, exists := fields["active"]; exists {
+		system.Active = fields["active"].(bool)
+	}
+	if _, exists := fields["sshRootEnabled"]; exists {
+		system.SshRootEnabled = fields["sshRootEnabled"].(bool)
+	}
+	if _, exists := fields["sshPassEnabled"]; exists {
+		system.SshPassEnabled = fields["sshPassEnabled"].(bool)
+	}
+	if _, exists := fields["version"]; exists {
+		system.Version = fields["version"].(string)
+	}
+	if _, exists := fields["agentVersion"]; exists {
+		system.AgentVersion = fields["agentVersion"].(string)
+	}
+	if _, exists := fields["allowPublicKeyAuthentication"]; exists {
+		system.AllowPublicKeyAuth = fields["allowPublicKeyAuthentication"].(bool)
+	}
+	if _, exists := fields["organization"]; exists {
+		system.Organization = fields["organization"].(string)
+	}
+	if _, exists := fields["created"]; exists {
+		system.Created = fields["created"].(string)
+	}
+	if _, exists := fields["arch"]; exists {
+		system.Arch = fields["arch"].(string)
+	}
+	if _, exists := fields["systemTimeZone"]; exists {
+		system.SystemTimezone = fields["systemTimezone"].(float64)
+	}
+	if _, exists := fields["allowSshPasswordAuthentication"]; exists {
+		system.AllowSshPasswordAuthentication = fields["allowSshPasswordAuthentication"].(bool)
+	}
+	if _, exists := fields["displayName"]; exists {
+		system.DisplayName = fields["displayName"].(string)
+	}
+	if _, exists := fields["modifySSHDConfig"]; exists {
+		system.ModifySSHDConfig = fields["modifySSHDConfig"].(bool)
+	}
+	if _, exists := fields["allowMultiFactorAuthentication"]; exists {
+		system.AllowMultiFactorAuthentication = fields["allowMultiFactorAuthentication"].(bool)
+	}
+	if _, exists := fields["hostname"]; exists {
+		system.Hostname = fields["hostname"].(string)
+	}
+
+	if _, exists := fields["sshdParams"]; exists {
+		system.SshdParams = getJCSSHDParamFromArray(fields["sshdParams"].([]interface{}))
+	}
+	if _, exists := fields["networkInterfaces"]; exists {
+		system.NetworkInterfaces = getJCNetworkInterfacesFromArray(fields["networkInterfaces"].([]interface{}))
+	}
+
+	if _, exists := fields["amazonInstanceID"]; exists {
+		system.AmazonInstanceID = fields["amazonInstanceID"].(string)
+	}
+}
+
+func getJCSystemsFromInterface(systemInt interface{}) []JCSystem {
+
+	var returnVal []JCSystem
+
+	recMap := systemInt.(map[string]interface{})
+
+	results := recMap["results"].([]interface{})
+
+	returnVal = make([]JCSystem, len(results))
+
+	for idx, result := range results {
+		getJCSystemFieldsFromInterface(result.(map[string]interface{}), &returnVal[idx])
+	}
+
+	return returnVal
+}
+
+// Executes a search by hostname via the JumpCloud API
+func (jc JCAPI) GetSystemByhostname(hostname string, withTags bool) ([]JCSystem, JCError) {
```

Comment:
Can we change the name here to:

GetSystemByHostname() (just change the lower-case 'h'?) for consistency?